src/blueapi/core/context.py

```diff
@@ -144,20 +146,41 @@ def device(self, device: Device, name: Optional[str] = None) -> None:
             raise TypeError(f"{device} is not a Bluesky compatible device")
 
         if name is None:
-            if isinstance(device, Readable) or isinstance(device, Flyable):
+            if isinstance(device, HasName):
                 name = device.name
             else:
                 raise KeyError(f"Must supply a name for this device: {device}")
 
         self.devices[name] = device
 
 
-def device_validators(ctx: BlueskyContext) -> Iterable[TypeValidatorDefinition]:
-    def get_device(name: str) -> Device:
-        device = ctx.find_device(name)
-        if device is None:
-            raise KeyError(f"Could not find a device named {name}")
-        return device
-
-    for proto in BLUESKY_PROTOCOLS:
-        yield TypeValidatorDefinition(proto, get_device)
+def plan_validators(
```

Comment:
I think this function may grow and change and possibly be split up and recombined in all sorts of ways as we add more fancy plan parameter processing. Could we rename it to `_plan_validators` to discourage it's use outside this module for now?